ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/operations/AttestationData.java

```diff
@@ -60,8 +60,8 @@ public AttestationData(UnsignedLong slot, AttestationData data) {
     this.slot = slot;
     this.index = data.getIndex();
     this.beacon_block_root = data.getBeacon_block_root();
-    this.source = data.getSource();
-    this.target = data.getTarget();
+    this.source = new Checkpoint(data.getSource());
+    this.target = new Checkpoint(data.getTarget());
```

Comment:
We could actually make `Checkpoint` immutable and not need to do this copy.  The setters are only used in a couple of tests which would could easily just use the `Checkpoint(UnsignedLong, Bytes32)` constructor instead.

Reply:
Agreed, but I'd rather do that in a separate PR: https://pegasys1.atlassian.net/secure/RapidBoard.jspa?rapidView=67&view=planning&selectedIssue=BC-168
I only made this small change here because I noticed that we have a potential bug here if we don't create a new Checkpoint object.